app/services/google_translation_service.rb

```diff
@@ -19,9 +20,49 @@ def translate(text, to)
     end
   end
 
+  def cache_listing_translations(listing_id, keys, translations)
+    listing = transform_translations_for_caching(listing_id, keys, translations)
+    if @cache.write("/ListingDetails/#{listing_id}", listing)
+      Rails.logger.info(
+        "Successfully cached listing translations for listing id: #{listing_id}",
+      )
+    else
+      Rails.logger.error(
+        "Error caching listing translations for listing id: #{listing_id}",
+      )
+    end
+    listing
+  end
+
+  private
+
   def parse_translations(results)
     return [results.text] unless results.is_a?(Array)
 
     results.map(&:text)
   end
+
+  def transform_translations_for_caching(listing_id, keys, translations)
+    listing = Force::ListingService.cached_listing(listing_id)
+    prev_cached_translations = listing[0][:translations] || {}
+    # keys can come from updated_values.keys in the event_subscriber_translate_service
+    # they will be in the same order as the translations because the translation service
+    # uses the values from that object this would make the assumption that every value
+    # comes back with 1 translation for each translation language (we would have a problem
```

Comment:
Even if google doesn't return a translation for a language, the api will still return a field so this is not a possibility